software/apps/audio_module/event_detector/dynArray.c

```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "dynArray.h"
+#include "_kiss_fft_guts.h"
+
+void initArray(Array *a, size_t initialSize) {
+    a->maxSize = initialSize;
+    a->used = 0;
+
+    a->size = a->maxSize;
+    printf("Init array size = %zu\n",a->size);
```

Comment:
This seems like a debug message? Perhaps better for _library code_ to do this in a `DEBUG_PRINTF` macro that can be turned off at compiled time? (As opposed to printfs from the actual application code, like in `main.c`)